File: static/code/sample/tethered/gdl-ldleds805-c.js
// In this sample:
// Show characters, dots, and numbers.

import pkg_serialusb from 'dlserialusb';
const {SerialUSB} = pkg_serialusb

import pkg_duelink from 'duelink';
const {DUELinkController} = pkg_duelink

let duelink = new DUELinkController(new SerialUSB());
await duelink.Connect();

// Sleep helper (ms)
function sleep(ms) {
    return new Promise(resolve => setTimeout(resolve, ms));
}

// Methods
async function DrawText(text) {
    await duelink.Engine.ExecuteCommand(`PrntStr("${text}")`);
}

async function DrawDot(pos, value) {
    await duelink.Engine.ExecuteCommand(`Dot(${pos},${value})`);
}

async function DrawNumber(number) {
    await duelink.Engine.ExecuteCommand(`Digit(${number})`);
}

async function Clear() {
    await duelink.Engine.ExecuteCommand("Clear(0)");
}

async function Show() {
    await duelink.Engine.ExecuteCommand("Show()");
}

(async () => {
    await DrawText("ABCDE");
    await Show();
    await sleep(1000);

    await DrawText("FGHIJ");
    await Show();
    await sleep(1000);

    await DrawText("KLMNO");
    await Show();
    await sleep(1000);

    await DrawText("PQRST");
    await Show();
    await sleep(1000);

    await DrawText("UVWXY");
    await Show();
    await sleep(1000);

    await DrawText("  Z  ");
    await DrawDot(0, 1);
    await DrawDot(1, 1);
    await DrawDot(3, 1);
    await DrawDot(4, 1);
    await Show();
    await sleep(1000);

    let n = 0;
    while (true) {
        await Clear();
        await DrawNumber(n % 100000);
        n++;
        await Show();
        await sleep(100);
    }
})();

File: static/code/sample/tethered/gdl-ldleds805-c.py
# In this sample:
# Show characters, dots, and numbers.

import time
from DUELink.DUELinkController import DUELinkController

availablePort = DUELinkController.GetConnectionPort()
duelink = DUELinkController(availablePort)

# method
def DrawText(text):
    duelink.Engine.ExecuteCommand(f'PrntStr("{text}")')

def DrawDot(pos, value):
    duelink.Engine.ExecuteCommand(f'Dot({pos},{value})')

def DrawNumber(number):
    duelink.Engine.ExecuteCommand(f'Digit({number})')

def Clear():
    duelink.Engine.ExecuteCommand("Clear(0)")

def Show():
    duelink.Engine.ExecuteCommand("Show()")


DrawText("ABCDE")
Show()
time.sleep(1)
DrawText("FGHIJ")
Show()
time.sleep(1)
DrawText("KLMNO")
Show()
time.sleep(1)
DrawText("PQRST")
Show()
time.sleep(1)
DrawText("UVWXY")
Show()
time.sleep(1)
DrawText("  Z  ")
DrawDot(0, 1)
DrawDot(1, 1)
DrawDot(3, 1)
DrawDot(4, 1)
Show()
time.sleep(1)

n = 0
while True:
    Clear()
    DrawNumber(n % 100000)
    n += 1
    Show()
    time.sleep(0.1)